.github/workflows/Build.yml

```diff
@@ -11,243 +11,263 @@ on:
       - uses: ./ci
 
-  #16
   vtr-optimized-linux:
-    runs-on: "ubuntu-20.04"
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
       PACKAGE: "pnr/vtr-optimized"
       OS_NAME: "linux"
     steps:
       - uses: actions/checkout@v3
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #17
   quicklogic-vtr-linux:
-    runs-on: "ubuntu-20.04"
-    needs: "vtr-linux"
+    needs: ["vtr-linux"]
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
       PACKAGE: "pnr/quicklogic-vtr"
       OS_NAME: "linux"
     steps:
-      # Skip if token isn't available (cross-repository PRs mainly)
-      - run: if [ "$ANACONDA_TOKEN" = "" ]; then echo "SKIP=true" >>$GITHUB_ENV; fi
       - uses: actions/checkout@v3
+      # Skip if token isn't available (cross-repository PRs mainly).
+      - run: if [ "$ANACONDA_TOKEN" = "" ]; then echo "SKIP=true" >>$GITHUB_ENV; fi
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #18
   quicklogic-vtr-gui-linux:
-    runs-on: "ubuntu-20.04"
-    needs: "vtr-gui-linux"
+    needs: ["vtr-gui-linux"]
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
       PACKAGE: "pnr/quicklogic-vtr-gui"
       OS_NAME: "linux"
     steps:
-      # Skip if token isn't available (cross-repository PRs mainly)
-      - run: if [ "$ANACONDA_TOKEN" = "" ]; then echo "SKIP=true" >>$GITHUB_ENV; fi
       - uses: actions/checkout@v3
+      # Skip if token isn't available (cross-repository PRs mainly).
+      - run: if [ "$ANACONDA_TOKEN" = "" ]; then echo "SKIP=true" >>$GITHUB_ENV; fi
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #19
   nextpnr-ice40-linux:
-    runs-on: "ubuntu-20.04"
     needs: ["icestorm-linux"]
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
       PACKAGE: "pnr/nextpnr/ice40"
       OS_NAME: "linux"
     steps:
       - uses: actions/checkout@v3
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #20
   nextpnr-ecp5-linux:
-    runs-on: "ubuntu-20.04"
     needs: ["prjtrellis-linux"]
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
       PACKAGE: "pnr/nextpnr/ecp5"
       OS_NAME: "linux"
     steps:
       - uses: actions/checkout@v3
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #21
   nextpnr-generic-linux:
-    runs-on: "ubuntu-20.04"
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
       PACKAGE: "pnr/nextpnr/generic"
       OS_NAME: "linux"
     steps:
       - uses: actions/checkout@v3
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #22
   nextpnr-xilinx-linux:
-    runs-on: "ubuntu-20.04"
+    needs: ["prjtrellis-linux"]
```

Comment:
Does `nextpnr-xilinx` really use `prjtrellis`? It doesn't seem so.

Reply:
Fixed.